server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -325,9 +325,29 @@ private static FieldCapabilitiesResponse merge(
         } else {
             collectResponseMap(responseMapBuilder, responseMap);
         }
+
+        // The merge method is only called on the primary coordinator for cross-cluster field caps, so we
+        // log relevant "5xx" errors that occurred in this 2xx response to ensure they are only logged once.
+        // These failures have already been deduplicated, before this method was called.
+        for (FieldCapabilitiesFailure failure : failures) {
+            LOGGER.warn(
+                "Field caps partial-results Exception for indices " + Arrays.toString(failure.getIndices()),
+                failure.getException()
+            );
+        }
         return new FieldCapabilitiesResponse(indices, Collections.unmodifiableMap(responseMap), failures);
     }
 
+    private static boolean shouldLogException(Exception e) {
```

Comment:
Where is this method called?

Reply:
Ack, thanks. I went through a number of rewrites of this PR and forgot to add the call back in to the point of logging. Fixed. It is now called from the logging code in the `merge` method.